src/libs/Middleware/SaveResponseInOnyx.js

```diff
@@ -16,19 +16,12 @@ const requestsToIgnoreLastUpdateID = ['OpenApp', 'ReconnectApp', 'GetMissingOnyx
 function SaveResponseInOnyx(requestResponse, request) {
     return requestResponse.then((response) => {
         // Make sure we have response data (i.e. response isn't a promise being passed down to us by a failed retry request and response undefined)
-        if (!response) {
+        if (!response && !request.successData && !request.failureData) {
             return;
         }
-        const onyxUpdates = response.onyxData;
-
-        // Sometimes we call requests that are successfull but they don't have any response or any success/failure data to set. Let's return early since
-        // we don't need to store anything here.
-        if (!onyxUpdates && !request.successData && !request.failureData) {
-            return Promise.resolve(response);
```

Comment:
I guess it's a similar question to @tgolen's, but who is using this promise and expecting the response? 🤔

Reply:
I think it's being used by the middlewares, more notably `SaveResponseInOnyx`. cc @danieldoglas since it seems like you added this check

Reply:
hmm we are in the middleware you referenced though?

Reply:
we return `Promise.resolve(response)` so we pass the response to the next middleware

Reply:
yes, there is no next middleware AFAICT

Reply:
so it's gonna be used here https://github.com/Expensify/App/blob/e14e9ee2724cf61de9bf4d1bb6dea2b1dcb96995/src/libs/Network/SequentialQueue.js#L71-L80

Reply:
Got it. Some of this queue pausing business is new to me. But that looks like a bug. On seeing that, I'd say we should either always return a response object or handle the possibility that the response is `undefined` when the `currentRequest` promise resolves.

Reply:
Cool, I think using an empty object as default might work here.